contracts/extensions/DeltaNeutralBasisTradingStrategyExtension.sol

```diff
@@ -0,0 +1,1456 @@
+                                                        // NOTE: Applies to both perpetual and dex trades.
+        uint256 twapCooldownPeriod;                     // Cooldown period required since last trade timestamp in seconds
+                                                        // NOTE: Applies to both perpetual and dex trades.
+    }
+
+    struct ExchangeSettings {
+        string exchangeName;                            // Name of the exchange adapter to be used for dex trade. This contract only supports UnisawpV3 trades.
+                                                        // So should be "UniswapV3ExchangeAdapterV2".
+        bytes buyExactSpotTradeData;                    // Bytes containing path and fixIn boolean which will be passed to TradeModule#trade to buy exact amount of spot asset.
+                                                        // Can be generated using UniswapV3ExchangeAdapterV2#generateDataParam
+        bytes sellExactSpotTradeData;                   // Bytes containing path and fixIn boolean which will be passed to TradeModule#trade to sell exact amount of spot asset
+                                                        // Can be generated using UniswapV3ExchangeAdapterV2#generateDataParam
+        bytes buySpotQuoteExactInputPath;               // Bytes containing UniswapV3 path to buy spot asset using exact amount of input asset (USDC). Will be passed to Quoter#getExactInput.
+        uint256 twapMaxTradeSize;                       // Max trade size in base assset base units. Always a positive number
+                                                        // NOTE: Applies to both perpetual and dex trades.
+        uint256 incentivizedTwapMaxTradeSize;           // Max trade size for incentivized rebalances in base asset units. Always a positive number
+                                                        // NOTE: Applies to both perpetual and dex trades.
+    }
+
+    struct IncentiveSettings {
+        uint256 etherReward;                            // ETH reward for incentivized rebalances
+        int256 incentivizedLeverageRatio;               // Leverage ratio for incentivized rebalances. Is a negative number lower than maxLeverageRatio.
+                                                        // E.g. -2x for ETH -1x.
+        uint256 incentivizedSlippageTolerance;          // Slippage tolerance percentage for incentivized rebalances
+                                                        // NOTE: Applies to both perpetual and dex trades.
+        uint256 incentivizedTwapCooldownPeriod;         // TWAP cooldown in seconds for incentivized rebalances
+                                                        // NOTE: Applies to both perpetual and dex trades.
+    }
+
+    /* ============ Events ============ */
+
+    // Todo: Emit spot and perp amounts separately? e.g. emit _chunkSpotRebalanceNotional, _totalSpotRebalanceNotional, _chunkPerpRebalanceNotional, _totalPerpRebalanceNotional.
```

Comment:
If it's not too much extra work seems like this would be worth while. Though would we expect this to always be the same?

Reply:
Yes, we are passing the same magnitude amount to both the spot and perp trade calls. And since they are invoked, there is no easy way to determine the output amount. But we can be sure that they wouldn't defer by more than a wei.
Although note that perp and spot trade notional amounts are always same in magnitued but opposite in signs.